machine/v1beta1/types_vsphereprovider.go

```diff
@@ -87,12 +87,61 @@ type NetworkSpec struct {
 	Devices []NetworkDeviceSpec `json:"devices"`
 }
 
+// AddressesFromPool is an IPAddressPool that will be used to create
+// IPAddressClaims for fulfillment by an external controller.
+type AddressesFromPool struct {
+	// group of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Group string `json:"group"`
+	// resource of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Resource string `json:"resource"`
+	// name of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
 // NetworkDeviceSpec defines the network configuration for a virtual machine's
 // network device.
 type NetworkDeviceSpec struct {
-	// NetworkName is the name of the vSphere network to which the device
-	// will be connected.
-	NetworkName string `json:"networkName"`
+	// networkName is the name of the vSphere network or port group to which the network
+	// device will be connected.
+	// +optional
+	NetworkName string `json:"networkName,omitempty"`
+
+	// gateway4 is the IPv4 gateway used by this device. Required if an IPv4 IP address
+	// is provided in IPAddrs.
+	// +kubebuilder:validation:Format=ipv4
```

Comment:
I don't think this works, but worth including for a note anyway, for documentation purposes